systemtest/test_jpscore.py

```diff
@@ -712,16 +713,48 @@ def test_door_closes_after_max_agents(tmp_path, env):
     )
     jpscore_driver.run()
 
-    # success = True
-
     max_agents_dict = read_max_agents(tmp_path / "inifile.xml")
     flow_dict = read_flow(tmp_path)
-    num_agents = read_num_agents(tmp_path / "inifile.xml")
-
     [max_agents_correct, measured_agents] = check_max_agents(
         flow_dict, max_agents_dict
     )
     assert max_agents_correct, "Wrong number of pedestrians passing the door"
     assert measured_agents == max_agents_dict
 
-    assert check_door_usage(flow_dict, num_agents), "Wrong door usage."
+    assert len(flow_dict[1].index) == 80
+    assert (
+        len(flow_dict[2].index) == 20
+    )  # trans id 2 is limited to 20 pedestrians
+    assert (
+        len(flow_dict[3].index) == 100
+    )  # all pedestrians should use main exit
+
+
+def test_door_flow_regulation(tmp_path, env):
+    """
+
+    :param tmp_path: working directory of test execution
+    :param env: global environment object
+    """
+    input_location = (
+        env.systemtest_path / "door_tests" / "test_flow_regulation"
+    )
+    copy_files(
+        sources=[
+            input_location / "geometry.xml",
+            input_location / "inifile.xml",
+            input_location / "events.xml",
+        ],
+        dest=tmp_path,
+    )
+    jpscore_driver = JpsCoreDriver(
+        jpscore_path=env.jpscore_path, working_directory=tmp_path
+    )
+    jpscore_driver.run()
+
+    max_agents_dict = read_max_agents(tmp_path / "inifile.xml")
+    outflow_dict = read_outflow_from_inifile(tmp_path / "inifile.xml")
+    starting_times_dict = read_starting_times(tmp_path / "events.xml")
+    data_dict = read_flow(tmp_path)
```

Comment:
Im not sure if I remember this correctly, but I think we discussed this and the conclusion was to not use the outflow files and instead compute the flow based on the trajectory. Esp. since we want to remove the statistics output eventually.

Reply:
done